server/model/monitor.js

```diff
@@ -521,29 +527,20 @@ class Monitor extends BeanModel {
                     bean.msg = `${res.status} - ${res.statusText}`;
                     bean.ping = dayjs().valueOf() - startTime;
 
-                    // Check certificate if https is used
-                    let certInfoStartTime = dayjs().valueOf();
+                    // Store certificate and check for expiry if https is used
                     if (this.getUrl()?.protocol === "https:") {
-                        log.debug("monitor", `[${this.name}] Check cert`);
-                        try {
-                            let tlsInfoObject = checkCertificate(res);
-                            tlsInfo = await this.updateTlsInfo(tlsInfoObject);
-
-                            if (!this.getIgnoreTls() && this.isEnabledExpiryNotification()) {
-                                log.debug("monitor", `[${this.name}] call checkCertExpiryNotifications`);
-                                await this.checkCertExpiryNotifications(tlsInfoObject);
-                            }
+                        // No way to listen for the `secureConnection` event, so we do it here
+                        const tlssocket = res.request.res.socket;
 
-                        } catch (e) {
-                            if (e.message !== "No TLS certificate in response") {
-                                log.error("monitor", "Caught error");
-                                log.error("monitor", e.message);
-                            }
+                        if (tlssocket) {
+                            tlsInfo.valid = tlssocket.authorized || false;
```

Comment:
@chakflying
I think this might be the source of https://github.com/louislam/uptime-kuma/issues/4693

Reply:
Indeed. I had that problem with 1.23.12, too and when I comment line 536 in the container and restart it, everything is back to normal as with 1.23.11.

I have done no deep-dive into the code, but I think that the root cause for the problem when connecting over a proxy could be that it is not the target connection TLS validity, but the connection socket to the proxy here. The latter can be unencrypted, despite the target URL being HTTPS - consider the case when you have a proxy "http://192.168.4.5:3128" and a monitoring target of "https://github.com".

Whatever the case, "tlsInfo.valid" can be "undefined" here in these cases and thus the assignment fails.